### DISCUSSION TITLE
WebRTC video stream rendering issue with Mapbox

### DISCUSSION THREAD

**Discussion by Grimsly** [rnmapbox/maps#2955] [Q&A]
I am trying to use a combination of [react-native-webrtc](https://github.com/react-native-webrtc/react-native-webrtc), [react-native-bottom-sheet](https://github.com/gorhom/react-native-bottom-sheet) and rnmapbox. There is a `ScrollView` in the bottom sheet and I noticed that the video stream viewer from the react-native-webrtc library is visible in the background above the bottom sheet if scrolled too far. Not sure if this could happen to any video component. This issue only happens to Android and not iOS strangely enough.

**Here is a screenshot in the iOS 12 simulator:**
<img src="https://github.com/react-native-webrtc/react-native-webrtc/assets/85096943/2ff3474b-f5f3-44be-8446-6c45da0a56a5" width="300" />

**Here is the screenshot in an Android phone (Samsung S10e):**
<img src="https://github.com/react-native-webrtc/react-native-webrtc/assets/85096943/e330c69a-c0fd-4888-a375-ccc730f3e83d" width="300" />

I used a sheet of white paper on my camera to contrast with the background. You can see in the Android screenshot that there is white sticking out of the bottom sheet when `RTCView` (the `react-native-webrtc` component) is scrolled past the top. I suspect the cause of the problem is Mapbox, because once I removed `MapView`, the stream viewer is not visible above the bottom sheet anymore. I tried setting the `MapView` `surfaceView` prop to `false` and the issue is still there.

I saw the rule, `no extra libraries` in relation to opening new issues, so I just decided to ask here instead if anyone has experienced this or has used react-native-webrtc with react-native-webrtc before? My `@rnmapbox/maps` version is `10.0.0-beta.44` by the way.

As far as I know, `react-native-webrtc` is using a [`SurfaceViewRenderer`](https://github.com/react-native-webrtc/react-native-webrtc/blob/c71ce8dd3e33993af4eb32675851a9551b84d40a/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java#L140) from a [snapshot of Chromium's WebRTC library](https://github.com/jitsi/webrtc).

**Comment by mfazekas** [comment]
@Grimsly you can try with `surfaceView={false}` on MapView.

**Reply by Grimsly** [reply]
I tried that. Still clips through for some reason.
